fix(curriculum): improve permutation generator description (#64723)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/lab-permutation-generator/66fe4f33a2cc9b33f4d5cd9b.md

In this lab, you will build a permutation generator that will take a string and return all possible permutations of the characters in the string. For example, the possible permutations of the string `cat` are `cat`, `cta`, `act`, `atc`, `tac`, and `tca`.

The recursive way of creating permutations of a string works by storing the fixed starting part of the string, and creating permutations of the rest.

An example can be written with a word like `machine`. The first round of creating permutations would be made fixing the `m` as the prefix of the string, and then creating permutations of the rest of the string `achine`.

> For the rest of the string, permutations continue in the same way, one letter is added to the prefix, maybe the `c`, so the prefix becomes `mc`, so to this then are concatenated all the permutations of `ahine`.

Reviewer: ```suggestion
For the rest of the string, permutations continue in the same way. One letter is added to the prefix, maybe the `c`, so the prefix becomes `mc`. Then, each of the permutations of `ahine` is concatenated to the prefix.
```
